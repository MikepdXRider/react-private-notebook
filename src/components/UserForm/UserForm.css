.form {
  display: grid;
  gap: 1rem;
  border: 2px dashed var(--gray);
  width: 400px;
  margin: 20px auto 10px;
  padding: 15px 20px 20px;
  text-align: left;
}

.form legend {
  text-align: center;
}

.formSection {
  display: grid;
  gap: 0.25rem;
}

.input {
  padding: 7px 10px;
  border: 1px solid var(--gray);
  font-size: 1.1rem;
  font-family: ProximaNova;
}
