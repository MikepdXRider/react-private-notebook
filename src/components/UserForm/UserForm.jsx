import React, { useState } from 'react';
import { useForm } from '../../hooks/useForm';

import styles from './UserForm.css';

export default function UserForm({
  className = '',
  label = 'Authenticate',
  onSubmit,
}) {
  const { formState, formError, handleFormChange, setFormError } = useForm({
    email: '',
    password: '',
  });
  const [loading, setLoading] = useState(false);

  const handleSubmit = async (event) => {
    event.preventDefault();
    const { email, password } = formState;

    try {
      if (!email || password.length < 8)
        throw new Error(
          'An email and password (with 8+ characters) are required.'
        );
      setLoading(true);
      await onSubmit(email, password);
    } catch (error) {
      setLoading(false);
      setFormError(error.message);
    }
  };

  return (
    <form className={className} onSubmit={handleSubmit}>
      <fieldset className={styles.form}>
        <legend>{label}</legend>
        <section className={styles.formSection}>
          <label htmlFor="email">Email</label>
          <input
            id="email"
            type="email"
            name="email"
            value={formState.email}
            className={styles.input}
            onChange={handleFormChange}
          />
        </section>
        <section className={styles.formSection}>
          <label htmlFor="password">Password</label>
          <input
            id="password"
            type="password"
            name="password"
            value={formState.password}
            className={styles.input}
            onChange={handleFormChange}
          />
        </section>
        <button type="submit" disabled={loading}>
          {loading ? 'Authenticating...' : label}
        </button>
        {formError && <p>{formError}</p>}
      </fieldset>
    </form>
  );
}
