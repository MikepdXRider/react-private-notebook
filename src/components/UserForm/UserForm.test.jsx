import { fireEvent, render, screen } from '@testing-library/react';
import UserForm from './UserForm';

it('should display a form for signing up/in', () => {
  const { container } = render(<UserForm />);
  expect(container).toMatchSnapshot();
});
