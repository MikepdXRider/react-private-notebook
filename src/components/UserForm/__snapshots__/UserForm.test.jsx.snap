// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a form for signing up/in 1`] = `
<div>
  <form
    class=""
  >
    <fieldset
      class="form"
    >
      <legend>
        Authenticate
      </legend>
      <section
        class="formSection"
      >
        <label
          for="email"
        >
          Email
        </label>
        <input
          class="input"
          id="email"
          name="email"
          type="email"
          value=""
        />
      </section>
      <section
        class="formSection"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          class="input"
          id="password"
          name="password"
          type="password"
          value=""
        />
      </section>
      <button
        type="submit"
      >
        Authenticate
      </button>
      
    </fieldset>
  </form>
</div>
`;
